vscode-smart-contracts/snippets.json

```diff
@@ -0,0 +1,88 @@
+{
+  "Smart contract receive-function": {
+    "prefix": ["screc"],
+    "body": [
+      "#[receive(",
+      "\tcontract = \"${1:my-contract}\",",
+      "\tname = \"${2:my-receive}\",",
+      "\tparameter = \"${3:MyParams}\",",
+      "\treturn_value = \"${4:MyReturnValue}\",",
+      "\terror = \"${5:MyContractError}\"",
+      ")]",
+      "fn $1_$2<S: HasStateApi>(",
+      "\tctx: &impl HasReceiveContext,",
+      "\thost: &mut impl HasHost<State<S>, StateApiType = S>",
+      ") -> Result<$4, $5> {",
+      "\tlet parameter: $3 = ctx.parameter_cursor().get()?;",
+      "\t$0",
+      "}"
+    ],
+    "description": "Receive function for a Concordium smart contract"
+  },
+  "Smart contract init-function": {
+    "prefix": ["scinit"],
+    "body": [
+      "#[init(",
+      "\tcontract = \"${1:my-contract}\",",
+      "\tparameter = \"${2:MyParams}\",",
+      "\terror = \"${3:MyContractError}\",",
+      "\tevent = \"${4:MyContractEvent}\"",
+      ")]",
+      "fn $1_init<S: HasStateApi>(",
+      "\tctx: &impl HasInitContext,",
+      "\tstate_builder: &mut StateBuilder<S>,",
+      ") -> Result<${5:MyContractState}<S>, $3> {",
+      "\tlet parameter: $2 = ctx.parameter_cursor().get()?;",
+      "\t$0",
+      "}"
+    ],
+    "description": "Init function for a Concordium smart contract"
+  },
+  "Smart contract state struct": {
+    "prefix": ["scstate"],
+    "body": [
+      "#[derive(Serial, DeserialWithState, StateClone)]",
+      "#[concordium(state_parameter = \"S\")]",
+      "struct ${1:MyContractState}<S> {",
+      "\t$0",
+      "}"
+    ],
+    "description": "State struct for a Concordium smart contract"
```

Comment:
Perhaps we should mention that the generic parameter Is only needed if you state* types in there.

Reply:
This description is not really shown to the user (only if they run a command for listing every snippet), so we can't really educate them here. 
Maybe the snippet should be without the generic?

Reply:
Ah okay.
Hmm, I am not sure about that.
On the one hand, we have this epic about removing the generics, which suggests you remove it from the snippet.
But on the other, the generic part is still being used and is harder to remember and type, so having a snippet for it is currently useful.

Reply:
I'll leave it as is for now.